docker/deploy/Dockerfile

```diff
@@ -4,17 +4,11 @@
 FROM ubuntu:xenial
 RUN apt-get update
 RUN apt-get -y install apt-utils
-RUN apt-get -y install sudo
-RUN apt-get install -y git cmake build-essential autoconf curl libtool python-dev python-numpy python-pip libboost-all-dev unzip graphviz
-RUN pip install ipython typing funcsigs subprocess32 protobuf colorama graphviz cloudpickle
-RUN adduser --gecos --ingroup ray-user --disabled-login --gecos ray-user
-RUN adduser ray-user sudo
-RUN sed -i "s|%sudo\tALL=(ALL:ALL) ALL|%sudo\tALL=NOPASSWD: ALL|" /etc/sudoers
-USER ray-user
-WORKDIR /home/ray-user
+RUN apt-get install -y git cmake build-essential autoconf curl libtool python-dev python-numpy python-pip libboost-all-dev unzip graphviz iputils-ping openssh-server
```

Comment:
Have a look at the main quilt Dockerfile.

Basically, what you want to do, is in a single giant ugly RUN apt-get all of the dependencies, build, and then remove all of build time dependencies.  That way we don't end up having giant layers in the container making everything bigger.